addons/portal/controllers/thread.py

```diff
@@ -0,0 +1,43 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import http
+from odoo.tools import html2plaintext
+from odoo.http import request
+from odoo.addons.mail.controllers import thread
+
+
+class ThreadController(thread.ThreadController):
+
+    @http.route()
+    def mail_message_post(self, thread_model, thread_id, post_data, context=None, **kwargs):
+        message = super().mail_message_post(thread_model, thread_id, post_data, context, **kwargs)
+        message_data = message.get("Message")
+        if (message_data := message_data and message_data[0]) and request.env[
+            "res.partner"
+        ]._get_portal_partner_from_context():
+            message = {
+                "default_message": html2plaintext(message_data.get("body")),
+                "default_message_id": message_data.get("id"),
+                "default_attachment_ids": message_data.get("attachments"),
+            }
+        return message
+
+    def _get_allowed_message_post_params(self):
+        post_params = super()._get_allowed_message_post_params()
+        if request.env["res.partner"]._get_portal_partner_from_context():
+            post_params.update(["author_id", "send_after_commit"])
+        return post_params
```

Comment:
There is still an issue with this override. It shouldn't exist.
This method is made to control user input parameters.
It is not made to control values generated from code (that we therefore generate safely).
The filtering should be done before or inside `_prepare_post_data` so that we don't need to add more keys here.